dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/DefaultIconService.java

```diff
@@ -57,141 +53,139 @@
-                .toList()
-                .stream(),
-            customIconStore.getIcons(iconOperationParams))
-        .collect(Collectors.toList());
+  public Set<Icon> getIcons(IconOperationParams params) {
+    return iconStore.getIcons(params);
   }
 
   @Override
   @Transactional(readOnly = true)
-  public long count(IconOperationParams iconOperationParams) {
-    return customIconStore.count(iconOperationParams);
+  public long count(IconOperationParams params) {
+    return iconStore.count(params);
   }
 
   @Override
   @Transactional(readOnly = true)
   public Icon getIcon(String key) throws NotFoundException {
-    if (defaultIcons.containsKey(key)) {
-      return defaultIcons.get(key);
+    Icon icon = iconStore.getIconByKey(key);
+    if (icon == null) {
+      throw new NotFoundException(String.format("Icon not found: %s", key));
     }
 
-    return getCustomIcon(key);
+    return icon;
   }
 
   @Override
   @Transactional(readOnly = true)
-  public CustomIcon getCustomIcon(String key) throws NotFoundException {
-    CustomIcon customIcon = customIconStore.getIconByKey(key);
-    if (customIcon == null) {
-      throw new NotFoundException(String.format("Icon not found: %s", key));
-    }
+  public Resource getIconResource(String key) throws NotFoundException {
 
-    return customIcon;
-  }
-
-  @Override
-  public Resource getDefaultIconResource(String key) throws NotFoundException {
-    if (defaultIcons.containsKey(key)) {
-      return new ClassPathResource(
-          String.format("%s/%s.%s", ICON_PATH, key, DefaultIcon.Icons.SUFFIX));
+    if (iconExists(key)) {
+      return new ClassPathResource(String.format("%s/%s.%s", ICON_PATH, key, DefaultIcon.SUFFIX));
     }
 
-    throw new NotFoundException(String.format("No default icon found with key %s.", key));
-  }
-
-  @Override
-  @Transactional(readOnly = true)
-  public Set<String> getKeywords() {
-    return Stream.concat(
-            defaultIcons.values().stream().map(Icon::getKeywords).flatMap(Arrays::stream),
-            customIconStore.getKeywords().stream())
-        .collect(Collectors.toSet());
+    throw new NotFoundException(String.format("No Icon found with key %s.", key));
   }
 
   @Override
   @Transactional(readOnly = true)
   public boolean iconExists(String key) {
-    return defaultIcons.get(key) != null || customIconStore.getIconByKey(key) != null;
+    return iconStore.getIconByKey(key) != null;
   }
 
   @Override
   @Transactional
-  public void addCustomIcon(CustomIcon customIcon) throws BadRequestException, NotFoundException {
-    validateIconDoesNotExists(customIcon.getKey());
-    FileResource fileResource = getFileResource(customIcon.getFileResourceUid());
-    UserDetails currentUserDetails = CurrentUserUtil.getCurrentUserDetails();
-    customIconStore.save(customIcon, fileResource, currentUserDetails);
-  }
+  public void addIcon(Icon icon) throws BadRequestException, NotFoundException, SQLException {
```

Comment:
Please add a `javax.annotation.Nonnull` to the `icon` parameter on the interface and implementation.